SYCL/ESIMD/api/functional/ctors/ctor_load_core.cpp

```diff
@@ -0,0 +1,256 @@
+//==------- ctor_load_core.cpp  - DPC++ ESIMD on-device test ---------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// REQUIRES: gpu, level_zero
+// XREQUIRES: gpu
+// TODO gpu and level_zero in REQUIRES due to only this platforms supported yet.
+// The current "REQUIRES" should be replaced with "gpu" only as mentioned in
+// "XREQUIRES".
+// UNSUPPORTED: cuda, hip
+// RUN: %clangxx -fsycl %s -fsycl-device-code-split=per_kernel -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+//
+// Test for simd load constructor.
+// The test uses reference data and different alignment flags. Invokes simd
+// constructors in different contexts with provided reference data and alignment
+// flag.
+// It is expected for destination simd instance to store a bitwise same data as
+// the reference one.
+
+#include "common.hpp"
+
+using namespace sycl::ext::intel::experimental::esimd;
+using namespace esimd_test::api::functional;
+
+// Descriptor class for the case of calling constructor in initializer context.
+struct initializer {
+  static std::string get_description() { return "initializer"; }
+
+  template <typename DataT, int NumElems, typename AlignmentT>
+  static void call_simd_ctor(const DataT *ref_data, DataT *const out,
+                             AlignmentT alignment) {
+    simd<DataT, NumElems> source_simd(ref_data, alignment);
+    simd<DataT, NumElems> simd_by_init = simd<DataT, NumElems>(source_simd);
```

Comment:
```suggestion
    simd<DataT, NumElems> simd_by_init = simd<DataT, NumElems>(ref_data, alignment);
```
Otherwise we would call copy constructor

Reply:
I've applied this comment [here](https://github.com/intel/llvm-test-suite/pull/769/commits/9de67432a3606acda7db15caab8cc4c7ef03c3f4#diff-3c39b599d2c35b37e91d3c2ef8b547db1db978189ce11ba0b9ca6e3fff47b5c7L36-R37)